test/releasers/java-yoshi.ts

```diff
@@ -734,4 +664,162 @@ describe('JavaYoshi', () => {
       )
     );
   });
+
+  it('creates a release PR against a feature branch', async () => {
+    // We stub the entire suggester API, asserting only that the
+    // the appropriate changes are proposed:
+    let expectedChanges = null;
+    let expectedOptions = null;
+    sandbox.replace(
+      suggester,
+      'createPullRequest',
+      (_octokit, changes, options): Promise<number> => {
+        expectedOptions = options;
+        expectedChanges = [...(changes as Map<string, object>)]; // Convert map to key/value pairs.
+        return Promise.resolve(22);
+      }
+    );
+    const versionsContent = readFileSync(
+      resolve(fixturesPath, 'versions.txt'),
+      'utf8'
+    );
+    const readmeContent = readFileSync(
+      resolve(fixturesPath, 'README.md'),
+      'utf8'
+    );
+    const pomContents = readFileSync(resolve(fixturesPath, 'pom.xml'), 'utf8');
+    const googleUtilsContent = readFileSync(
+      resolve(fixturesPath, 'GoogleUtils.java'),
+      'utf8'
+    ).replace(/\r\n/g, '\n');
+    const graphql = JSON.parse(
+      readFileSync(resolve(fixturesPath, 'commits-yoshi-java.json'), 'utf8')
+    );
+    const req = nock('https://api.github.com')
```

Comment:
I'm wondering if it would be worth moving some of this logic to `sinon`, like [here](https://github.com/googleapis/release-please/blob/master/test/releasers/rust.ts#L147) I've been finding that it feels a bit more maintainable than enumerating all of the GitHub endpoints we'll be interacting with.

Reply:
I like this a lot. Let me do it in a separate, follow-up PR to avoid muddying up this PR.